dwave/preprocessing/include/dwave/presolve.h

```diff
@@ -262,7 +262,7 @@ void Presolver<bias_type, index_type, assignment_type>::apply() {
         while (c < model_.num_constraints()) {
             auto& constraint = model_.constraint_ref(c);
 
-            if (constraint.num_variables() == 0) {
+            if (constraint.num_variables() == 0 && !constraint.is_soft()) {
```

Comment:
I think we can just add the penalty to the offset of the objective!

Reply:
excellent point

Reply:
I didn't end up adding to the offset, but I do remove the empty soft constraint. See note in the code for why